components/application-mgt/org.wso2.carbon.identity.application.common/src/main/java/org/wso2/carbon/identity/application/common/model/SpTrustedAppMetadata.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.common.model;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.axis2.databinding.annotation.IgnoreNullElement;
+
+import java.io.Serializable;
+import java.util.Iterator;
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
+ * Trusted app metadata of an application.
+ */
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlRootElement(name = "SpTrustedAppMetadata")
+public class SpTrustedAppMetadata implements Serializable {
+
+    private static final long serialVersionUID = 8734858467750306551L;
+    private static final String ANDROID_PACKAGE_NAME = "AndroidPackageName";
+    private static final String ANDROID_THUMBPRINTS = "AndroidThumbprints";
+    private static final String APPLE_APP_ID = "AppleAppId";
+    private static final String IS_FIDO_TRUSTED = "IsFidoTrusted";
+
+    @IgnoreNullElement
+    @XmlElement(name = ANDROID_PACKAGE_NAME)
+    private String androidPackageName;
+
+    @IgnoreNullElement
+    @XmlElement(name = ANDROID_THUMBPRINTS)
+    private String androidThumbprints;
```

Comment:
Lets have this as an array. At the data level its fine to keep as it is. Having this as an array eliminates the need to perform splitting in multiple places

Reply:
Let the DAO level do the needed transformation